common/metrics/otel_metrics_handler.go

```diff
@@ -87,30 +113,61 @@ func (omp *otelMetricsHandler) Counter(counter string) CounterIface {
 	}
 
 	return CounterFunc(func(i int64, t ...Tag) {
-		option := metric.WithAttributes(tagsToAttributes(omp.tags, t, omp.excludeTags)...)
+		option := metric.WithAttributeSet(omp.makeSet(t))
 		c.Add(context.Background(), i, option)
 	})
 }
 
+func (omp *otelMetricsHandler) getGaugeAdapter(gauge string) (*gaugeAdapter, error) {
+	if v, ok := omp.gauges.Load(gauge); ok {
+		return v.(*gaugeAdapter), nil
+	}
+	adapter := &gaugeAdapter{
+		values: make(map[attribute.Distinct]gaugeValue),
+	}
+	if v, wasLoaded := omp.gauges.LoadOrStore(gauge, adapter); wasLoaded {
+		return v.(*gaugeAdapter), nil
+	}
+
+	opts := addOptions(omp, gaugeOptions{
+		metric.WithFloat64Callback(adapter.callback),
+	}, gauge)
+	_, err := omp.provider.GetMeter().Float64ObservableGauge(gauge, opts...)
```

Comment:
This looks a little funny--could you comment what it's doing? I'm guessing that just making this call has the side effect of registering the metric with this set of tags in otel?

Reply:
yup

Reply:
(except to be clear, there are no tags here, the tags come later)